core/src/thrift/tachyon.thrift

```diff
@@ -261,19 +262,22 @@ service WorkerService {
     throws (1: FileDoesNotExistException eP, 2: SuspectedFileSizeException eS,
       3: BlockInfoException eB)
 
-  string getDataFolder()
+  void cancelBlock(1: i64 userId, 2: i64 blockId)
 
-  string getUserTempFolder(1: i64 userId)
+  string getBlockLocation(1: i64 userId, 2: i64 blockId, 3: i64 initialBytes)
+    throws (1: OutOfSpaceException eP, 2: FileAlreadyExistException eS)
 
   string getUserUfsTempFolder(1: i64 userId)
 
-  void lockBlock(1: i64 blockId, 2: i64 userId) // Lock the file in memory while the user is reading it.
+  string lockBlock(1: i64 blockId, 2: i64 userId) // Lock the file in memory while the user is reading it.
+    throws (1: FileDoesNotExistException eP) // the path of the file locked is returned
 
-  void returnSpace(1: i64 userId, 2: i64 returnedBytes)
+  bool promoteBlock(1: i64 userId, 2: i64 blockId)
 
-  bool requestSpace(1: i64 userId, 2: i64 requestBytes)   // Should change this to return i64, means how much space to grant.
+  bool requestSpace(1: i64 userId, 2: i64 blockId, 3: i64 requestBytes)
```

Comment:
It would be better to have explanations of these methods like the one you wrote in docx in JIRA. e.g. for this one:


Return true if the worker successfully allocate space for the block on block’s location, if there is no space, OutOfSpaceException will be thrown, if there is no information of the block on worker, FileDoesNotExistException will be thrown.Return true if the worker successfully allocate space for the block on block’s location, if there is no space, OutOfSpaceException will be thrown, if there is no information of the block on worker, FileDoesNotExistException will be thrown.